File: backend/dump_sqlite.js
const sqlite3 = require('sqlite3').verbose();
const fs = require('fs');
const path = require('path');

const dbPath = path.join(__dirname, 'lib_interno.sqlite');
const db = new sqlite3.Database(dbPath);

const tables = ['clients', 'order_services', 'suppliers_registry'];
const data = {};

let pending = tables.length;

tables.forEach(table => {
    db.all(`SELECT * FROM ${table}`, [], (err, rows) => {
        if (err) {
            console.error(`Error reading ${table}:`, err.message);
            data[table] = [];
        } else {
            data[table] = rows;
            console.log(`Read ${rows.length} rows from ${table}`);
        }
        pending--;
        if (pending === 0) {
            fs.writeFileSync(path.join(__dirname, 'dump_data.json'), JSON.stringify(data, null, 2));
            console.log('Done! Data saved to dump_data.json');
            db.close();
        }
    });
});

File: backend/dump_sqlite.py
import sqlite3
import json
import os

db_path = 'lib_interno_correct.sqlite'
tables = ['clients', 'order_services', 'suppliers_registry']
data = {}

conn = sqlite3.connect(db_path)
conn.row_factory = sqlite3.Row
cursor = conn.cursor()

for table in tables:
    try:
        cursor.execute(f"SELECT * FROM {table}")
        rows = [dict(row) for row in cursor.fetchall()]
        data[table] = rows
        print(f"Read {len(rows)} rows from {table}")
    except Exception as e:
        print(f"Error reading {table}: {e}")
        data[table] = []

with open('dump_data.json', 'w', encoding='utf-8') as f:
    json.dump(data, f, ensure_ascii=False, indent=2)

conn.close()
print("Done! Data saved to dump_data.json")
